src/Generator/EnumGenerator/EnumGenerator.php

```diff
@@ -61,7 +58,7 @@ private function retrieveType(): string
             return '';
         }
 
-        return ': ' . $this->cases->getBackedType();
+        return ': ' . $this->cases->type;
```

Comment:
We might be better off checking for the reverse here as this class is not marked internal, and has no native type check on the `$cases` variable passed to the constructor.

```suggestion
    if ($this->cases instanceof BackedCases) {
        return ': ' . $this->cases->type;
    }
    
    return '';
```

Reply:
63739216703f88ba8fbbe7fec6d8656babf1fbdd